src/main/java/org/broadinstitute/consent/http/service/dao/DatasetServiceDAO.java

```diff
@@ -27,31 +31,113 @@ public DatasetServiceDAO(Jdbi jdbi, DatasetDAO datasetDAO) {
+        // files
+        executeInsertFilesForDataset(handle, uploadedFiles, userId, datasetId);
+
+        return datasetId;
+    }
+
     public List<DatasetProperty> synchronizeDatasetProperties(Integer datasetId, List<DatasetProperty> properties) throws SQLException {
         jdbi.useHandle(
                 handle -> {
                     // By default, new connections are set to auto-commit which breaks our rollback strategy.
                     // Turn that off for this connection. This will not affect existing or new connections and
                     // only applies to the current one in this handle.
                     handle.getConnection().setAutoCommit(false);
-                    // 1. Generate inserts for missing dictionary terms
-                    // 2. Generate inserts for new dataset properties
-                    // 3. Generate updates for existing dataset properties
-                    // 4. Generate deletes for outdated dataset properties
-                    List<Update> updates = new ArrayList<>(generateDictionaryInserts(handle, properties));
-                    // We need to know existing properties for all property operations
-                    Set<DatasetProperty> existingProps = datasetDAO.findDatasetPropertiesByDatasetId(datasetId);
-                    updates.addAll(generatePropertyInserts(handle, datasetId, properties, existingProps));
-                    updates.addAll(generatePropertyUpdates(handle, datasetId, properties, existingProps));
-                    updates.addAll(generatePropertyDeletes(handle, properties, existingProps));
-                    updates.forEach(Update::execute);
+                    executeSynchronizeDatasetProperties(handle, datasetId, properties);
                     handle.commit();
                 }
         );
         return datasetDAO.findDatasetPropertiesByDatasetId(datasetId).stream().toList();
     }
 
+    private void executeInsertFilesForDataset(Handle handle, List<FileStorageObject> files, Integer userId, Integer datasetId) {
+        FileStorageObjectDAO fileStorageObjectDAO = handle.attach(FileStorageObjectDAO.class);
+        for (FileStorageObject file : files) {
+            fileStorageObjectDAO.insertNewFile(
+                    file.getFileName(),
+                    file.getCategory().getValue(),
+                    file.getBlobId().toGsUtilUri(),
+                    file.getMediaType(),
+                    datasetId.toString(),
+                    userId,
+                    Instant.now()
+            );
+        }
+    }
+
     // Helper methods to generate Dictionary inserts
+    private void executeSynchronizeDatasetProperties(Handle handle, Integer datasetId, List<DatasetProperty> properties) {
+        // 1. Generate inserts for missing dictionary terms
+        // 2. Generate inserts for new dataset properties
+        // 3. Generate updates for existing dataset properties
+        // 4. Generate deletes for outdated dataset properties
```

Comment:
While we are improving this code, perhaps it's worth moving these comments above the line of code that is executed?